src/openforms/registrations/contrib/stuf_zds/plugin.py

```diff
@@ -477,6 +478,58 @@ def get_config_actions(self):
             ),
         ]
 
+    def update_registration_with_confirmation_email(
+        self, submission: Submission, options: RegistrationOptions
+    ) -> dict:
+        logger.info("stuf_zds_finalise_registration_started", options=options)
+        res = get_last_confirmation_email(submission)
+        if res is None:
+            logger.warning(
+                "stuf_zds_finalise_registration_aborted",
+                reason="confirmation_email_not_sent",
+            )
+            raise RegistrationFailed("Confirmation email was not sent")
```

Comment:
probably we should identify if a confirmation e-mail is supposed to have been sent in the first place. It'd be nice if we can rely on `submission.confirmation_email_sent` for this, but then we probably need to update `send_confirmation_email` to set this flag if there are no `to_emails`.